src/main/java/org/stellar/sdk/Asset.java

```diff
@@ -42,11 +42,11 @@ public static Asset fromXdr(org.stellar.sdk.xdr.Asset xdr) {
         String assetCode4 = Util.paddedByteArrayToString(xdr.getAlphaNum4().getAssetCode().getAssetCode4());
         KeyPair issuer4 = KeyPair.fromXdrPublicKey(
                 xdr.getAlphaNum4().getIssuer().getAccountID());
-        return new AssetTypeCreditAlphaNum4(assetCode4, issuer4);
+        return new AssetTypeCreditAlphaNum4(assetCode4, issuer4.getAccountId());
```

Comment:
The problem here is that `xdr.getAlphaNum4().getIssuer().getAccountID()` can be invalid pubket and `KeyPair.fromXdrPublicKey` will try to create `EdDSAPublicKeySpec` instance throwing an exception.